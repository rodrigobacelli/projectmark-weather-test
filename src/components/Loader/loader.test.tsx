import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Loader } from './';

describe('components/NoResults', () => {
  it('should accept classNames', () => {
    const result = render(<Loader className="loader-className" />);

    expect(
      result.container.querySelector('.loader-className')
    ).toBeInTheDocument();
  });

  it('should render the NoResults Component', () => {
    const result = render(<Loader className="loader-className" />);

    expect(result).toMatchSnapshot();
  });
});
