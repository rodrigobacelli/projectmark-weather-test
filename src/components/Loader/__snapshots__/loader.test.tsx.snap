// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/NoResults > should render the NoResults Component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full flex items-center justify-center loader-className"
      >
        <svg
          class="animate-spin text-gray-700 dark:text-gray-400"
          fill="none"
          style="width: 40px; height: 40px;"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 24c0 11.0457 8.9543 20 20 20s20-8.9543 20-20S35.0457 4 24 4"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="4"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full flex items-center justify-center loader-className"
    >
      <svg
        class="animate-spin text-gray-700 dark:text-gray-400"
        fill="none"
        style="width: 40px; height: 40px;"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 24c0 11.0457 8.9543 20 20 20s20-8.9543 20-20S35.0457 4 24 4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="4"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
